### ISSUE TITLE
General solving pipeline

### ISSUE THREAD

**Issue by MarcelBruckner** [Brucknem/Generalization-Ideas-in-Deep-Learning#6]
A pipeline for solving neural nets is needed.
Basic capabilities are:
- [x] Pass a model to optimize
- [x] Switch between solving algorithms (SGD, Adam, ...)
- [x] Adjust parameters of solver
- [x] Train solver
- [x] Verbose output of training loss
- [x] Verbose output of training and validation accuracy

**Comment by MarcelBruckner** [comment]
Added really nice dynamic printing of [Bokeh](https://docs.bokeh.org/en/latest/) plots for accuracies and loss.

**Comment by MarcelBruckner** [comment]
Saving the whole solver is missing.
